crates/relayer/src/chain/axon.rs

```diff
@@ -838,10 +838,80 @@ impl ChainEndpoint for AxonChain {
         Ok((sequence.into(), None))
     }
 
-    // TODO do we need to implement this?
-    fn query_txs(&self, _request: QueryTxRequest) -> Result<Vec<IbcEventWithHeight>, Error> {
-        warn!("axon query_txs() not support");
-        Ok(vec![])
+    fn query_txs(&self, request: QueryTxRequest) -> Result<Vec<IbcEventWithHeight>, Error> {
+        let events = match request {
+            QueryTxRequest::Client(QueryClientEventRequest {
+                query_height: _,
+                event_id: _,
+                client_id,
+                consensus_height,
+            }) => {
+                // return at most one update client event
+                let block = self
+                    .rt
+                    .block_on(self.client.get_block(consensus_height.revision_height()))
+                    .map_err(|e| Error::other_error(e.to_string()))?;
+                let Some(block) = block else {
+                    return Ok(Vec::new());
+                };
+                let filter = Filter::new()
+                    .address(self.config.contract_address)
+                    .at_block_hash(block.hash.unwrap());
+                let logs = self
+                    .rt
+                    .block_on(self.client.get_logs(&filter))
+                    .map_err(|e| Error::other_error(e.to_string()))?;
+
+                logs.into_iter()
+                    .filter_map(|log| {
+                        let height = {
+                            let number = log.block_number.expect("no block number").as_u64();
+                            Height::from_noncosmos_height(number)
+                        };
+                        let tx_hash: [u8; 32] = log.transaction_hash.expect("no tx hash").into();
+                        let event =
+                            OwnableIBCHandlerEvents::decode_log(&log.into()).expect("parse log");
+                        match &event {
+                            OwnableIBCHandlerEvents::UpdateClientFilter(filter)
```

Comment:
we have CreateClient event

Reply:
https://github.com/synapseweb3/forcerelay/blob/main/crates/relayer/src/chain/cosmos.rs#L1702

This interface only returns update client event..